### PR TITLE
Added Demo App changes for Mono/Stereo Audio Mode and Updated SDK Tests

### PR DESCRIPTION
**Pull request by nggandhi**
### Issue #, if available:
N/A

### Description of changes:
**SDK**
* Updated `AudioMode` enum with values `Stereo48K`, `Mono48K`, `Mono16K` and `NoAudio`
> * The  `Stereo48K` corresponds to 48KHz Stereo Audio
> * The  `Mono48K` corresponds to 48KHz Mono Audio
> * The  `Mono16K` corresponds to 16KHz Mono Audio
> * The  `NoAudio` corresponds to No Audio
* The *Stereo/48KHz* Audio will now be the default Audio Mode, if none is explicitly specified when starting the audio session. Historically, the default and the only Audio Mode has been *Mono/16KHz* Audio.
* The Audio Mode specified when starting the audio session will be passed onto the Audio Client when calling the `startSession` API.
* Updated Unit Tests
* Updated README and CHANGELOG.

**Demo App**
* Removed the Join without Audio button, and instead added a Spinner in the `HomeActivity` to pick the Audio Mode when joining a meeting.

#### Unit test coverage
* Class coverage: 82%
* Line coverage: 67%

#### Manual test cases (add more as needed):
* [x] Join meeting
* [x] Leave meeting
* [x] Rejoin meeting
* [x] Send audio
* [x] Receive audio
* [x] See active speaker indicator when speaking
* [x] Mute/Unmute self
* [x] See local mute indicator when muted
* [x] See remote mute indicator when other is muted
* [x] See audio video events
* [x] See signal strength changes
* [x] See media metrics received
* [x] See roster updates when remote attendees join / leave the meeting
* [x] Enable local video
* [x] See local video tile
* [x] See remote video tile
* [x] Switch camera
* [x] See remote screen sharing content with attendee name
* [x] Pause remote video tile
* [x] Resume remote video tile
* [x] First time audio permissions
* [x] First time video permissions

Following are the manual testing scenarios performed by Audio Team (@zhemitu-amzn)
* Tested joining meeting with Stereo/48KHz
* Tested joining meeting with Mono/48KHz
* Tested joining meeting with Mono/16KHz
* Tested joining meeting with No Audio

*Endpoint Used for Testing: https://wmvhgew0bi.execute-api.us-east-1.amazonaws.com/Prod*

#### Screenshots, if available:

AudioModeSpinner

AudioModeSpinner_Expanded

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/internal/audio/DefaultAudioClientController.kt [hunk 1]
```diff
@@ -152,21 +166,26 @@ class DefaultAudioClientController(
...
+            }
             val res = audioClient.startSessionV2(
                 AudioClient.XTL_DEFAULT_TRANSPORT,
                 host,
                 port,
                 joinToken,
                 meetingId,
                 attendeeId,
-                AudioClient.kCodecOpusLow,
-                AudioClient.kCodecOpusLow,
```
Comment (ziyiz-amzn) [inline amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/internal/audio/DefaultAudioClientController.kt:163]: Is this function signature changed? why this 2 params are removed?

File: app/src/main/java/com/amazonaws/services/chime/sdkdemo/activity/HomeActivity.kt [hunk 2]
```diff
@@ -72,15 +76,12 @@ class HomeActivity : AppCompatActivity() {
 
         meetingEditText = findViewById(R.id.editMeetingId)
         nameEditText = findViewById(R.id.editName)
+        audioMode = findViewById(R.id.audioModeSpinner)
         authenticationProgressBar = findViewById(R.id.progressAuthentication)
         debugSettingsViewModel = ViewModelProvider(this).get(DebugSettingsViewModel::class.java)
 
+        audioMode?.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, audioModes)
```
Comment (ziyiz-amzn) [inline app/src/main/java/com/amazonaws/services/chime/sdkdemo/activity/HomeActivity.kt:83]: A better name for `simple_list_item_1`?

File: app/src/main/res/layout/activity_home.xml [hunk 3]
```diff
@@ -42,21 +42,19 @@
...
 
     <Button
-        android:id="@+id/buttonContinueWithoutAudio"
+        android:id="@+id/buttonContinue"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="@string/join_without_audio"
-        android:contentDescription="@string/main_continue_without_audio"
+        android:text="@string/join_meeting"
+        android:contentDescription="@string/main_continue"
```
Comment (hokyungh) [inline app/src/main/res/layout/activity_home.xml:57]: Will the integration tests updated with new UI?

File: amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/internal/audio/DefaultAudioClientController.kt (hunk 1 above)
Comment (nggandhi) [reply amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/internal/audio/DefaultAudioClientController.kt:163]: The function signature is changed in the media sdk. @zhemitu-amzn can you explain why?

File: app/src/main/java/com/amazonaws/services/chime/sdkdemo/activity/HomeActivity.kt (hunk 2 above)
Comment (nggandhi) [reply app/src/main/java/com/amazonaws/services/chime/sdkdemo/activity/HomeActivity.kt:83]: This is Android internal layout for simple text list item. Notice the *android.R* i.e Android-Internal, it's not our custom layout.

File: app/src/main/res/layout/activity_home.xml (hunk 3 above)
Comment (nggandhi) [reply app/src/main/res/layout/activity_home.xml:57]: No new integration tests were added for Join Without Audio when it was a separate button. The integration test will continue to work for regular Join Meeting flow.

Comment (ziyiz-amzn) [reply app/src/main/res/layout/activity_home.xml:57]: Will adding new integ test for this audio be a task you have planned?

File: amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/internal/audio/DefaultAudioClientController.kt (hunk 1 above)
Comment (zhemitu-amzn) [reply amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/internal/audio/DefaultAudioClientController.kt:163]: Yes. From the inside audio tincan code, a new function signature was added. The purpose was to support future stereo audio API and the current audio session API the same time.

File: amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/audiovideo/audio/AudioMode.kt [hunk 4]
```diff
@@ -15,9 +15,19 @@ enum class AudioMode(val value: Int) {
     NoAudio(0),
 
     /**
-     * The default audio mode with single audio channel.
+     * The mono audio mode with single audio channel and 16KHz.
```
Comment (zhemitu-amzn) [inline amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/audiovideo/audio/AudioMode.kt]: nit: 16kHz sampling rate

File: amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/audiovideo/audio/AudioMode.kt [hunk 5]
```diff
@@ -15,9 +15,19 @@ enum class AudioMode(val value: Int) {
...
      */
-    Mono(1);
+    Mono16K(1),
+
+    /**
+     * The mono audio mode with single audio channel and 48KHz.
```
Comment (zhemitu-amzn) [inline amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/audiovideo/audio/AudioMode.kt]: nit: 48kHz sampling rate

File: amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/audiovideo/audio/AudioMode.kt [hunk 6]
```diff
@@ -15,9 +15,19 @@ enum class AudioMode(val value: Int) {
...
+     */
+    Mono48K(2),
+
+    /**
+     * The stereo audio mode with two audio channels and 48KHz.
```
Comment (zhemitu-amzn) [inline amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/audiovideo/audio/AudioMode.kt]: nit: 48kHz sampling rate

File: app/src/main/res/layout/activity_home.xml (hunk 3 above)
Comment (nggandhi) [reply app/src/main/res/layout/activity_home.xml:57]: We haven't planned that task.

**Review by ziyiz-amzn — APPROVED** [review]

**Review by hokyungh — APPROVED** [review]
Could you try and see if integration tests passed?

**Review by zhemitu-amzn — APPROVED** [review]

**Pull request merged**